### ISSUE TITLE
annotation overview graphical display

### ISSUE THREAD

**Issue by mellybelly** [geneontology/amigo#296]
Maybe such a thing exists elsewhere, and I am sure you have already considered this. 

It would be great if you could look at all the annotations in a graph view, with the nodes that are annotated over the closure indicated in some fashion. @cmungall has done something similar for diseases here: https://github.com/monarch-initiative/monarch-disease-ontology/blob/master/reports/genes/CDKN2A.png
where the nodes annotated are highlighted in black (example more complex here because the nodes also come from different sources).

Essentially what you want to know is at quick glance, what are the key aspects of the functionality coming from different annotations, sources, organisms; and where does a candidate annotation fit into the existing graph view of the annotation set.

**Comment by cmungall** [comment]
We used to have this, for comparing annotations to a set of orthologous genes (@kltm - this was the Reference Genome display, based on what Mary did. With the little pie-charts next to each box, remember?)

Not worth resurrecting the old code I imagine. In fact this could probably be bolted on using the existing very generic API here: http://wiki.geneontology.org/index.php/AmiGO_2_Manual:_Visualize (I think there are undocumented features for coloring)

**Comment by mellybelly** [comment]
We'd ideally have something generic that we can use for other graphs too. @doctorbud might be useful for ontology navigation work also

**Comment by kltm** [comment]
@cmungall that "internal" visualize API does indeed have coloring and some other stuff, so no problem there: http://wiki.geneontology.org/index.php/AmiGO_2_Manual:_JSON#Node_Description_Format
(really, a wrapper around some of the stable bits of GraphViz).

It's been a while since the refgenome stuff was active, but I think the code still lives just fine (AmiGO::SVGRewrite). I think the main issue there with reusing that is that we no longer have the data setup to drive it.

**Comment by ValWood** [comment]
Bear in mind that for the average well annotated gene these graphs get a bit unmanageable. We link to an equivalent from Ensembl from the gene pages (top right corner of each ontology section where it says 'ontology graph). In fact Ensembl are removing these graphs from the  next release. I know our users haven't found them to be particularly useful.
They could be better if  they had useful zoom but for most gene products they are a bit unwieldy.

**Comment by cmungall** [comment]
I agree, and this is probably even more true for mammals than for pombe, 
due to proliferation of annotations from IMPs where we have the same 
gene active in multiple different organ systems

That said, there is opportunity here for some visual optimizations, e.g. 
eliminating non-informative nodes that don't form MRCAs, rolling up 
saturated parts of the graph up to ancestors like 'development', etc.

**Comment by ValWood** [comment]
Yes we were talking with FB curators on Friday and saying how useful it would be to have visualization with branch specific collapsing etc. Although this discussion  was in the context of enrichment and slims it would be very helpful for these views too.
